src/main/java/uk/gov/pay/connector/gatewayaccount/model/GatewayAccountEntity.java

```diff
@@ -183,6 +187,20 @@ public String getGatewayName() {
         return gatewayName;
     }
 
+    public String getGatewayNameFromGatewayAccountCredentials() {
+        List<GatewayAccountCredentialsEntity> gatewayAccountCredentialsEntities = getGatewayAccountCredentials();
+        if (gatewayAccountCredentialsEntities.size() == 1) {
+            GatewayAccountCredentialsEntity entity = gatewayAccountCredentialsEntities.get(0);
+            return entity.getPaymentProvider();
+        }
+
+        return gatewayAccountCredentialsEntities.stream()
+                .filter(entity -> entity.getState() == ACTIVE)
+                .max(Comparator.comparing(GatewayAccountCredentialsEntity::getActiveStartDate))
+                .get()
+                .getPaymentProvider();
```

Comment:
Can we throw exception instead of `get()` and returning ?
If there are no credentials at all or no creds with ACTIVE state this will throw Index out of bounds exception

```suggestion
Optional<GatewayAccountCredentialsEntity> mayBeGatewayAccountCredential = gatewayAccountCredentialsEntities
                .stream()
                .filter(entity -> entity.getState() == ACTIVE)
                .max(Comparator.comparing(GatewayAccountCredentialsEntity::getActiveStartDate));

        return mayBeGatewayAccountCredential.map(GatewayAccountCredentialsEntity::getPaymentProvider)
                .orElseThrow(() -> new WebApplicationException(serviceErrorResponse(
                        format("Credentials not found for gateway account %s", getId()))));
```